soroban-env-host/src/host.rs

```diff
@@ -438,6 +438,23 @@ impl Host {
         res
     }
 
+    /// Pushes an artificial [`Frame`], runs a closure, and then pops the frame,
+    /// rolling back if the closure returned an error. Returns the result that
+    /// the closure returned (or any error caused during the frame push/pop).
+    /// Used for testing.
+    #[cfg(any(test, feature = "testutils"))]
+    pub fn with_artificial_test_contract_frame<F>(
```

Comment:
I suspect there is a more appropriate name for this function, but it is not appearing to me at this moment.

Reply:
Is the "artificial" necessary? `with_test_contract_frame` sounds descriptive enough to me.

Reply:
I was undecided on that. I can remove it.